gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2324,15 +2514,31 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
         if( poFieldDefn->GetWidth() == 1 )
         {
             nType = NC_CHAR;
-            status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1,
+
+            if(!m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation